### DISCUSSION TITLE
Change part scheduling according to shipments instead of sales orders

### DISCUSSION THREAD

**Discussion by HannesMorre** [inventree/InvenTree#5180] [Q&A] [answered]
Since it is possible to split a sales order into multiple shipments, it seems logical to me that part scheduling should be adjusted according to these shipments instead of sales orders. See the example below:

There is a sales order with target date 30/08/2023. This sales order holds 3 line items (articles A, B and C) each 50 pcs.
Articles A and B are fully allocated to shipment 1 (which is scheduled for 15/07/2023)
25 pcs of article C are allocated to shipment 2 (scheduled for 01/08/2023)
25 pcs of article C are allocated to shipment 3 (scheduled for 15/08/2023)

In the part scheduling of these parts (A, B and C) we will see a drop on 30/08/2023. I think this should be 15/07/2023 for articles A and B, and for article C this drop should be split up on 01/08/2023 and 15/08/2023 instead of 30/08/2023.

Or am I doing something wrong?

**Comment by SchrodingersGat** [comment, marked as answer]
@HannesMorre fair point, the scheduling could take this into account - where items are allocated against shipments, and the shipment has a scheduled date. I'll transfer this into an issue for someone to tackle
